waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/client/pool/MetaStoreClientPool.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (C) 2016-2017 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.client.pool;
+
+import java.util.NoSuchElementException;
+
+import org.apache.commons.pool2.KeyedObjectPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hotels.bdp.waggledance.api.WaggleDanceException;
+import com.hotels.bdp.waggledance.api.model.AbstractMetaStore;
+import com.hotels.bdp.waggledance.client.CloseableThriftHiveMetastoreIface;
+
+/**
+ * Simple delegate to get rid of generics and add unchecked {@link #borrowObjectUnchecked(AbstractMetaStore)},
+ * {@link #returnObjectUnchecked(AbstractMetaStore, CloseableThriftHiveMetastoreIface) methods.
+ */
+public class MetaStoreClientPool implements KeyedObjectPool<AbstractMetaStore, CloseableThriftHiveMetastoreIface> {
```

Comment:
did you check this project? -> https://github.com/aloha-app/thrift-client-pool-java
I'm not sure if it could be used instead of implementing our own pool

Reply:
I had a quick look, but went for commons pool implementation. It all ends up being the same tbh, you need to create some factory for the clients and then configure you're pool accordingly.